shelly/src/main/java/au/com/jtribe/shelly/Dial.java

```diff
@@ -0,0 +1,72 @@
+package au.com.jtribe.shelly;
+
+import android.content.Intent;
+import android.net.Uri;
+import android.os.Build;
+
+/**
+ * Represents a phone dial to a number using Intent.ACTION_DIAL
+ */
+public class Dial {
+
+  private String phoneNumber;
+
+  public Dial() {
+  }
+
+  /**
+   * Adds a phone number to the Dial object that will be dialed using the intent
+   *
+   * @param number Phone number that will be dialed
+   * @return Dial object this method was called on for method chaining.
+   */
+  public Dial number(String number) {
```

Comment:
Annotate the method with `@CheckResult` and `@NonNull`
Parameter with `@NonNull`